lib/Config.js

```diff
@@ -67,13 +86,76 @@ class Config {
             });
         }
 
-        assert(typeof config.regions === 'object',
-               'bad config: the list of regions is mandatory');
-        assert(Object.keys(config.regions).every(
-               r => typeof r === 'string' && config.regions[r] instanceof Array
-               && config.regions[r].every(e => typeof e === 'string')),
-               'bad config: regions must be a set of {region: [endpoints]}');
-        this.regions = config.regions;
+        // legacy
+        if (config.regions !== undefined) {
+            assert(typeof config.regions === 'object',
+                'bad config: the list of regions is mandatory');
+            assert(Object.keys(config.regions).every(
+                r => typeof r === 'string' && config.regions[r] instanceof Array
+                && config.regions[r].every(e => typeof e === 'string')),
+                'bad config: regions must be a set of {region: [endpoints]}');
+            this.regions = config.regions;
+        }
+
+        this.locationConstraints = {};
+        if (config.locationConstraints !== undefined) {
+            assert(typeof config.locationConstraints === 'object',
+                'bad config: locationConstraints must be an object');
+            Object.keys(config.locationConstraints).forEach(l => {
+                assert(typeof config.locationConstraints[l].type === 'string' &&
+                    typeof config.locationConstraints[l].information
+                        === 'object',
+                    'bad config: locationConstraints.type and ' +
+                    'locationConstraints.information are mandatory- type ' +
+                    'must be a string and information must be an object');
+                this.locationConstraints[l] = config.locationConstraints[l];
+                this.locationConstraints[l].type =
+                    config.locationConstraints[l].type;
+                const info = config.locationConstraints[l].information;
+                const stringFields = [
+                    'region',
+                    'awsEndpoint',
+                    'bucketName',
+                    'credentialsProfile',
+                ];
+                stringFields.forEach(field => {
+                    if (info.field !== undefined) {
```

Comment:
I actually think doing it this way is more concise because otherwise I'd have to generate a new array containing all the fields for every locationConstraint, and then loop through that, right?